assets/js/blocks/product-collection/inspector-controls/on-sale-control.tsx

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { BlockEditProps } from '@wordpress/blocks';
+import {
+	ToggleControl,
+	// @ts-expect-error Using experimental features
+	// eslint-disable-next-line @wordpress/no-unsafe-wp-apis
+	__experimentalToolsPanelItem as ToolsPanelItem,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { ProductCollectionAttributes } from '../types';
+
+const OnSaleControl = (
+	props: BlockEditProps< ProductCollectionAttributes >
```

Comment:
A continuation from my comment above, if we are using just the `attributes` prop and you think we can only pass it to this component, we should probably get rid of the BlockEditProps and create an interface just with the props that are going to be used

Reply:
Thanks for your suggestion, @thealexandrelara. You're right, `OnSaleControl` component is indeed only using the attributes prop currently. Passing only the required props will make our component more explicit and could prevent unnecessary re-renders. I'll make the necessary adjustments in separate PR because I have created a few more PRs which are based on this branch. Appreciate your keen observation. 🙌🏻